configapi/api_slice_mgmt.go

```diff
@@ -198,14 +200,27 @@ func NetworkSlicePostHandler(c *gin.Context, msgOp int) bool {
 	logger.ConfigLog.Infof("url: %v ", req.URL)
 	procReq := req.Body.(configmodels.Slice)
 
+	for _, gnb := range procReq.SiteInfo.GNodeBs {
+		if !isValidName(gnb.Name) {
+			err := fmt.Errorf("invalid gNB name `%s` in Network Slice %s. Name needs to match the following regular expression: %s", gnb.Name, sliceName, NAME_PATTERN)
+			logger.ConfigLog.Error(err.Error())
+			return err
+		}
+		castedGnbTac := fmt.Sprint(gnb.Tac)
+		if !isValidGnbTac(castedGnbTac) {
+			err := fmt.Errorf("invalid TAC %s for gNB %s in Network Slice %s. TAC must be a numeric string within the range [1, 16777215]", castedGnbTac, gnb.Name, sliceName)
+			logger.ConfigLog.Error(err.Error())
+			return err
+		}
+	}
 	slice := procReq.SliceId
 	logger.ConfigLog.Infof("network slice: sst: %v, sd: %v", slice.Sst, slice.Sd)
 
 	group := procReq.SiteDeviceGroup
 	slices.Sort(group)
-	slices.Compact(group)
+	group = slices.Compact(group)
 	logger.ConfigLog.Infof("number of device groups %v", len(group))
-	for i := 0; i < len(group); i++ {
+	for i := range group {
 		logger.ConfigLog.Infof("device groups(%v) - %v", i+1, group[i])
 	}
```

Comment:
```suggestion
	for i, g := range group {
		logger.ConfigLog.Infof("device groups(%v) - %v", i+1, g)
	}
```
since you are already changing to use range.